src/services/wallet/walletService.ts

```
@@ -54,10 +63,60 @@ export async function ensureWalletForUser(
+
+/**
+ * Validate and set stellar address for a user (e.g., for imported wallets).
+ * This is the ONLY function that should be used to set stellarAddress externally.
+ */
+export async function setStellarAddressForUser(
+  userId: string,
+  stellarAddress: string,
+): Promise<void> {
+  // Strict validation
+  assertValidStellarAddress(stellarAddress);
+  
```

ESLint

[error] Delete `··`

(prettier/prettier)